### DISCUSSION TITLE
External queries not working in a CRON automation

### DISCUSSION THREAD

**Discussion by R2bEEaton** [Budibase/budibase#6474] [Help] [answered]
Hello!

I am having this issue where my CRON triggered automation does not seem to get any data back from an external query. However, when I test the query, it works just fine. I have also verified that the CRON is triggering by sending myself an email with the query results (which ends up blank). Was wondering if anyone had any guidance on this.

Thanks!

![image](https://user-images.githubusercontent.com/34921506/175958200-0c25d789-7c54-41dc-a875-42ae359fa012.png)

**Comment by melohagan** [comment]
Hey @R2bEEaton 

Just to confirm - is your app published?

**Reply by R2bEEaton** [reply]
@melohagan Hi, thanks for your quick reply! It is published. When I publish the app, the emails start coming in every minute but the query response is missing from where I put it in the email body (for debug purposes).

Edit: I know it's not a momentary problem with the external data connector because I have the email outputs for the past hour and it hasn't shown up once. Also, an internal "query rows" works just fine. Neither have filters (I know filters are broken right now).

**Reply by melohagan** [reply]
Would you be able to share a screenshot of your **Send SMTP** step please? Maybe something is configured incorrectly - although in this simple example if the test is passing you would think it should work.

**Reply by R2bEEaton** [reply]
Attached are pictures of my automation, along with two emails. The one missing data after the External Data Connector was run via cron, while the one that includes data was run via Automation testing. I have also found that it runs correctly when activated by a button for example.

![image](https://user-images.githubusercontent.com/34921506/176159302-d6dfeaba-bd06-4fac-ab83-3030527817e9.png)
![image](https://user-images.githubusercontent.com/34921506/176158348-bb28ca70-e193-405d-ab1e-aec69738f88a.png)
![image](https://user-images.githubusercontent.com/34921506/176158469-16e0d20a-31e5-4686-8afa-bf26eb4aa5d8.png)

**Comment by melohagan** [comment, marked as answer]
@R2bEEaton I find it interesting that it works if you replace the CRON with an app action trigger.

What I would try is instead of doing every minute, trying executing at a time in the near future to see if that works.
As it's every minute, maybe the cron is too frequent for it to handle?

**Reply by R2bEEaton** [reply]
That's a thought, thanks! I will hopefully try that tomorrow and report back. But then of course there's the fact that no matter how quickly I use the app action, it works every time there... Anyways, will try it.

**Reply by R2bEEaton** [reply]
Hello, sorry for the very delayed response -
Setting it to a specific time worked. I don't think it was to do with frequency though due to the SQL server because frequent manual activations of the query worked just fine. Fortunately I've worked around this and it is no longer an issue for me, and I don't think we've narrowed it down to where the problem comes from so there's not much else to do here.

**Reply by melohagan** [reply]
Thank you for getting back and letting people know what worked for you!
